### PR TITLE
Add share button

### PR DESCRIPTION
**Pull request by fstanis**
This PR adds a share button (using [`navigator.share`](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/share)) to PWA Directory.

### CODE REVIEW

File: public/js/ui/share-button.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+  _getTitle() {
+    const pwaName = this._window.document.querySelector('#pwa-name');
+    if (!pwaName) {
+      return 'PWA Directory';
+    }
+    return pwaName.innerText.trim();
+  }
+
+  share() {
```
Comment (andreban) [inline public/js/ui/share-button.js]: If I understood correctly, the share button is always available, but share fails if the browser doesn't support it. Would it be possible to simply hide the button if that's the case?

Comment (fstanis) [reply public/js/ui/share-button.js]: It's disabled in `_init` if unavailable - I could hide it instead, but was worried about a "flash" of going from visible to invisible. Thoughts?

Comment (fstanis) [reply public/js/ui/share-button.js]: Good point, done.

**Comment by juliantoledo** [comment]
Very cool, do you have a screenshot of how it looks?

**Comment by fstanis** [comment]
wxukaatwygg

The dialog itself is the platform-native one, e.g. on Android it will be the [share chooser](https://developer.android.com/training/sharing/send).

**Pull request merged**